### PR TITLE
{lyn10540} Fixing some cases where the return value for GraphObjectProxy Invoke

### PR DESCRIPTION
**Pull request by jackalbe**
Fixing some cases where the return value for GraphObjectProxy Invoke where a pointer, reference, or big value could be returned. Updated the tests to reflect the changes.

Signed-off-by: Allen Jackson <23512001+jackalbe@users.noreply.github.com>

### CODE REVIEW

**Review by AMZN-stankowi — APPROVED** [review]

**Review by amzn-mike — APPROVED** [review]

File: Code/Tools/SceneAPI/SceneCore/Containers/GraphObjectProxy.cpp [hunk 1]
```diff
@@ -292,8 +292,28 @@ namespace AZ
...
                 {
                     returnBehaviorValue.Set(*behaviorMethod->GetResult());
-                    returnBehaviorValue.m_value =
-                        returnBehaviorValue.m_tempData.allocate(returnBehaviorValue.m_azRtti->GetTypeSize(), 16);
+                    const size_t typeSize = returnBehaviorValue.m_azRtti->GetTypeSize();
+
+                    if (returnBehaviorValue.m_traits & BehaviorParameter::TR_POINTER)
+                    {
+                        // we need value to pointer be pointer to a pointer
+                        void* valuePointer = returnBehaviorValue.m_tempData.allocate(2 * sizeof(void*), 16);
```
[the remark this answers is not in the document]
Comment (jackalbe) [reply Code/Tools/SceneAPI/SceneCore/Containers/GraphObjectProxy.cpp:300]: that is what Lua did, so I just re-added it and used the exact same comment

[the remark this answers is not in the document]
Comment (jackalbe) [reply Code/Tools/SceneAPI/SceneCore/Containers/GraphObjectProxy.cpp:300]: The script context is using the tempAllocator here, I am not sure why this needs three new classes in order to store a pointer.

[the remark this answers is not in the document]
Comment (lumberyard-employee-dm) [reply Code/Tools/SceneAPI/SceneCore/Containers/GraphObjectProxy.cpp:300]: I mean add a  comment to explain what is going on here, a new struct isn't necessarily needed..
As it stands this block is nothing but a magic number to anyone reading it without any knowledge of the ScriptContext

[the remark this answers is not in the document]
Comment (jackalbe) [reply Code/Tools/SceneAPI/SceneCore/Containers/GraphObjectProxy.cpp:300]: ah, will do!

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**